Boost unused variable warning

## Description

Since #3001 was merged, compiling anything downstream yields the following warning for me:

```
In file included from stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/beta.hpp:1721,
                 from stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/binomial.hpp:15,
                 from stan/lib/stan_math/stan/math/prim/fun/choose.hpp:7,
                 from stan/lib/stan_math/stan/math/prim/fun.hpp:46,
                 from stan/lib/stan_math/stan/math/rev/fun/multiply.hpp:7,
                 from stan/lib/stan_math/stan/math/rev/fun/elt_multiply.hpp:9,
                 from stan/lib/stan_math/stan/math/rev/fun.hpp:56,
                 from stan/lib/stan_math/stan/math/rev.hpp:12,
                 from stan/lib/stan_math/stan/math.hpp:19,
                 from stan/src/stan/model/model_header.hpp:4,
                 from ../../ml/stanc3/double-reject.hpp:2:
stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp: In instantiation of ‘boost::math::detail::temme_root_finder<T>::temme_root_finder(T, T) [with T = double]’:
stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp:304:7:   required from ‘T boost::math::detail::temme_method_2_ibeta_inverse(T, T, T, T, T, const Policy&) [with T = double; Policy = boost::math::policies::policy<boost::math::policies::pole_error<boost::math::policies::errno_on_error>, boost::math::policies::overflow_error<boost::math::policies::errno_on_error>, boost::math::policies::promote_float<false>, boost::math::policies::promote_double<false>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy>]’
stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp:615:48:   required from ‘T boost::math::detail::ibeta_inv_imp(T, T, T, T, const Policy&, T*) [with T = double; Policy = boost::math::policies::policy<boost::math::policies::pole_error<boost::math::policies::errno_on_error>, boost::math::policies::overflow_error<boost::math::policies::errno_on_error>, boost::math::policies::promote_float<false>, boost::math::policies::promote_double<false>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy>]’
stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp:992:30:   required from ‘boost::math::tools::promote_args_t<T1, T2, T3, T4> boost::math::ibeta_inv(T1, T2, T3, T4*, const Policy&) [with T1 = double; T2 = double; T3 = double; T4 = double; Policy = boost::math::policies::policy<boost::math::policies::overflow_error<boost::math::policies::errno_on_error>, boost::math::policies::pole_error<boost::math::policies::errno_on_error>, boost::math::policies::promote_double<false>, boost::math::policies::digits2<0>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy>; boost::math::tools::promote_args_t<T1, T2, T3, T4> = double]’
stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp:1023:20:   required from ‘boost::math::tools::promote_args_t<RT1, RT2, A> boost::math::ibeta_inv(RT1, RT2, RT3, const Policy&) [with RT1 = double; RT2 = double; RT3 = double; Policy = boost::math::policies::policy<boost::math::policies::overflow_error<boost::math::policies::errno_on_error>, boost::math::policies::pole_error<boost::math::policies::errno_on_error>, boost::math::policies::promote_double<false>, boost::math::policies::digits2<0>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy>; boost::math::tools::promote_args_t<RT1, RT2, A> = double]’
stan/lib/stan_math/stan/math/prim/fun/inv_inc_beta.hpp:32:32:   required from here
stan/lib/stan_math/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp:29:15: warning: unused variable ‘x_extrema’ [-Wunused-variable]
   29 |       const T x_extrema = 1 / (1 + a);
      |       
```

Are there any objections to removing that line from the boost file? @andrjohns @SteveBronder 
#### Current Version:
v4.8.1

--- a/lib/boost_1.84.0/STAN_CHANGES
+++ b/lib/boost_1.84.0/STAN_CHANGES
@@ -1,0 +1,4 @@
+This file documents changes done for the stan-math project
+
+- Added a `#ifndef BOOST_DISABLE_ASSERTS` to boost/math/special_functions/detail/ibeta_inverse.hpp
+  to silence a warning about an unused variable.

--- a/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp
+++ b/lib/boost_1.84.0/boost/math/special_functions/detail/ibeta_inverse.hpp
@@ -26,8 +26,10 @@ template <class T>
 struct temme_root_finder
 {
    temme_root_finder(const T t_, const T a_) : t(t_), a(a_) {
+      #ifndef BOOST_DISABLE_ASSERTS
       const T x_extrema = 1 / (1 + a);
       BOOST_MATH_ASSERT(0 < x_extrema && x_extrema < 1);
+      #endif
    }
 
    boost::math::tuple<T, T> operator()(T x)
@@ -121,7 +123,7 @@ T temme_method_1_ibeta_inverse(T a, T b, T z, const Policy& pol)
       x = 0;
    else if (x > 1)
       x = 1;
-   
+
    BOOST_MATH_ASSERT(eta * (x - 0.5) >= 0);
 #ifdef BOOST_INSTRUMENT
    std::cout << "Estimating x with Temme method 1: " << x << std::endl;
@@ -406,7 +408,7 @@ T temme_method_3_ibeta_inverse(T a, T b, T p, T q, const Policy& pol)
 
    // Early exit for cases with numerical precision issues.
    if (cross == 0 || cross == 1) { return cross; }
-   
+
    x = tools::newton_raphson_iterate(
       temme_root_finder<T>(u, mu), x, lower, upper, policies::digits<T, Policy>() / 2);
 #ifdef BOOST_INSTRUMENT
@@ -842,7 +844,7 @@ T ibeta_inv_imp(T a, T b, T p, T q, const Policy& pol, T* py)
       else if(pow(p, 1/a) < 0.5)
       {
 #ifndef BOOST_NO_EXCEPTIONS
-         try 
+         try
          {
 #endif
             x = pow(p * a * boost::math::beta(a, b, pol), 1 / a);
@@ -863,7 +865,7 @@ T ibeta_inv_imp(T a, T b, T p, T q, const Policy& pol, T* py)
       {
          // model a distorted quarter circle:
 #ifndef BOOST_NO_EXCEPTIONS
-         try 
+         try
          {
 #endif
             y = pow(1 - pow(p, b * boost::math::beta(a, b, pol)), 1/b);

